solana-labs/solana-web3.js
# refactor(experimental): an implementation of the fetch shim that tree-shakes

refactor(experimental): an implementation of the fetch shim that tree-shakes
## Summary

By creating a subpackage that builds browser/node versions of the `fetch` shim, we can practically keep the `node-fetch` import _out_ of the browser builds of downstream packages.


## Test Plan

```
cd packages/rpc-transport
pnpm i
pnpm turbo compile:js
```

* Inspect the browser package; observe no `node-fetch` import
* Inspect the node package; observe `import fetchImpl from \`node-fetch\``
